### PR TITLE
#6 : Create login page in Angular

### PR DESCRIPTION
**Pull request by newklearz**
Create login component
     Create auth component for future routing of Login, Register and ForgetPassword

### CODE REVIEW

**Comment by newklearz** [comment]
Login page design:
Capture

File: project-management-application-web/src/app/auth/components/login/login.component.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<link href="//maxcdn.bootstrapcdn.com/bootstrap/4.1.1/css/bootstrap.min.css" rel="stylesheet" id="bootstrap-css">
+<script src="//maxcdn.bootstrapcdn.com/bootstrap/4.1.1/js/bootstrap.min.js"></script>
+<script src="//cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
+<div class="container login-container">
```
Comment (pionnegru) [inline project-management-application-web/src/app/auth/components/login/login.component.html:4]: what's the difference between `container` and  `login-container` class?

File: project-management-application-web/src/app/auth/components/login/login.component.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<link href="//maxcdn.bootstrapcdn.com/bootstrap/4.1.1/css/bootstrap.min.css" rel="stylesheet" id="bootstrap-css">
+<script src="//maxcdn.bootstrapcdn.com/bootstrap/4.1.1/js/bootstrap.min.js"></script>
+<script src="//cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
```
Comment (pionnegru) [inline project-management-application-web/src/app/auth/components/login/login.component.html:3]: these should go in index.html

File: project-management-application-web/src/app/auth/components/login/login.component.css [hunk 3]
```diff
@@ -0,0 +1,34 @@
+.login-container{
+  margin: 20%;
+}
+.login-form-1{
```
Comment (pionnegru) [inline project-management-application-web/src/app/auth/components/login/login.component.css:4]: why append `-1`?

File: project-management-application-web/src/app/auth/components/login/login.component.css [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  border: none;
+  cursor: pointer;
+}
+.login-form-1 .btnSubmit{
+  font-weight: 600;
+  color: #fff;
+  background-color: #0062cc;
+}
+
+.login-form-1 .ForgetPwd{
```
Comment (pionnegru) [inline project-management-application-web/src/app/auth/components/login/login.component.css:30]: class should not be with capitalization

File: project-management-application-web/src/app/auth/components/login/login.component.css [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+}
+.login-form-1 h3{
+  text-align: center;
+  color: #333;
+}
+
+.login-container form{
+  padding: 10%;
+}
+.btnSubmit
```
Comment (pionnegru) [inline project-management-application-web/src/app/auth/components/login/login.component.css:16]: lowercase

File: project-management-application-web/src/app/auth/components/login/login.component.html [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+<div class="container login-container">
+  <div class="row">
+    <div class="col-md-6 login-form-1">
+      <h3>Welcome to PMA</h3>
```
Comment (pionnegru) [inline project-management-application-web/src/app/auth/components/login/login.component.html:7]: why does the login component has this information? you should break the app in to components and views. in this case, login should be just the component and the login view should put together other details

File: project-management-application-web/src/app/auth/components/login/login.component.html [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      <form>
+        <div class="form-group">
+          <input type="text" class="form-control" placeholder="Your Username *" value=""/>
+        </div>
+        <div class="form-group">
+          <input type="password" class="form-control" placeholder="Your Password *" value=""/>
```
Comment (pionnegru) [inline project-management-application-web/src/app/auth/components/login/login.component.html:13]: you should also add required attribute , not just the visual `*`.  also, maybe a toggle password visibility would be handy

File: project-management-application-web/src/app/auth/components/login/login.component.html [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+        </div>
+        <div class="form-group">
+          <input type="submit" class="btnSubmit" value="Login"/>
```
Comment (pionnegru) [inline project-management-application-web/src/app/auth/components/login/login.component.html:16]: you have a submit action but the form doesn't handle this, I guess its work in progress?

File: project-management-application-web/src/app/auth/auth-routing.module.ts [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import { NgModule } from '@angular/core';
+import { RouterModule, Routes } from '@angular/router';
+
+const routes: Routes = [
+  {path:'',redirectTo:'login', pathMatch: 'full'}
+];
+
+@NgModule({
+  imports: [RouterModule.forChild(routes)],
```
Comment (pionnegru) [inline project-management-application-web/src/app/auth/auth-routing.module.ts:9]: where is the root? :)

**Review by pionnegru — CHANGES_REQUESTED** [review]
for styling you should consider [BEM](http://getbem.com/introduction/) its hard to read the intension. also, you can break the css in *theme.css that handles colors and decorations from *.css that just positions things

**Review by Epurashu — APPROVED** [review]
lgtm

File: project-management-application-web/src/app/auth/components/login/login.component.ts [hunk 10]
```diff
@@ -3,13 +3,22 @@ import { Component, OnInit } from '@angular/core';
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
-  styleUrls: ['./login.component.css']
+  styleUrls: ['./login.component.css','./login.themes.component.css'],
 })
 export class LoginComponent implements OnInit {
+  text: String ='password';
```
Comment (pionnegru) [inline project-management-application-web/src/app/auth/components/login/login.component.ts:9]: better go with a `boolean`  something like `showPassword` and you just toggle the value

File: project-management-application-web/src/app/auth/auth-routing.module.ts [hunk 11]
```diff
@@ -2,7 +2,7 @@ import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
 
 const routes: Routes = [
-  {path:'',redirectTo:'login', pathMatch: 'full'}
+  {path:'',redirectTo:'/login', pathMatch: 'full'}
```
Comment (pionnegru) [inline project-management-application-web/src/app/auth/auth-routing.module.ts:5]: being a child route, this should render a component not redirect from child to a main root

**Review by Epurashu — APPROVED** [review]

**Pull request merged**